openpype/pipeline/workfile/path_resolving.py

```diff
@@ -534,3 +538,36 @@ def create_workdir_extra_folders(
         fullpath = os.path.join(workdir, subfolder)
         if not os.path.exists(fullpath):
             os.makedirs(fullpath)
+
+
+def get_last_workfile_representation(
+    project_name: str,
+    asset_name: str,
+    task_name: str,
+) -> dict:
+    """Get last published workfile representation.
+
+    Args:
+        project_name(str): Project name.
+        asset_name(str): Asset/Shot name.
+        task_name(str): Task name.
+
+    Returns:
+        dict: Last workfile representation.
+    """
+    return max(
```

Comment:
I'll let others pitch in, but I think this is more on the complex side, and probably would split it in more lines.

Reply:
Agree - the `max` call will exhaust the `filter` generator anyway so we could just as well do a list comprehension before this:
```python
representations = [repre for repre in representations if repre["context"].get("version") is not None]
```

Then this would just become:
```python
return max(representations, key=lambda r: r["context"]["version"])
```

Reply:
100% agree. Please do readable code. Also consider that `context_filters` won't work in AYON, and because AYON is our current main goal it is not good idea to put this to core functions.